modules/widget-visibility/widget-conditions.php

```diff
@@ -235,6 +235,50 @@ public static function widget_conditions_options_echo( $major = '', $minor = ''
 				</optgroup>
 				<?php
 			break;
+
+			case 'single_post':
+				// Get Post object.
+				$post_types = array(
+					'post' => get_post_type_object( 'post' )
+				);
+
+				$custom_post_types = get_post_types(
+					array(
+						'public'   => true,
+						'_builtin' => false,
+					),
+					'objects'
+				);
+
+				// Add "Post" object to other post types.
+				$post_types = array_merge( $post_types, $custom_post_types );
+
+				foreach ( $post_types as $post_type ) {
+				?>
+					<optgroup label="<?php printf( esc_attr( '%s :', 'jetpack' ), $post_type->label ); ?>">
```

Comment:
Almost there! You can remove the `jetpack` textdomain here, since it won't really be used. The label will be translated beforehand, when the post type is registered. If you remove the semi-colon (which isn't really necessary in my opinion since it's a dropdown), then you're left with something simple:

```php
<optgroup label="<?php echo esc_attr( $post_type->label );  ?>">
```